api/client/swarm/init.go

```diff
@@ -36,7 +37,8 @@ func newInitCommand(dockerCli *client.DockerCli) *cobra.Command {
 	}
 
 	flags := cmd.Flags()
-	flags.Var(&opts.listenAddr, "listen-addr", "Listen address")
+	flags.Var(&opts.listenAddr, flagListenAddr, "Listen address")
+	flags.StringVar(&opts.listenInterface, flagListenInterface, "", "Listen on an interface's address")
```

Comment:
> If I want to use --listen-interface with a different port then I'd have to pass a "bogus" --listen-addr.

Yeah, I agree it's not ideal.

> Alternative: --listen-addr eth0:4242

I would rather not overload `--listen-addr` like this. `eth0` could be a valid hostname as well.

I thought of `--listen-interface eth0:4242`, but subinterfaces can contain `:` characters.

Any ideas?